### ISSUE TITLE
id attribute value of active elements is unique 

### ISSUE THREAD

**Issue by Brynman** [dequelabs/axe-core#3363]
Product: << axe-core 

Expectation: << Not flat duplicate IDs on elements with display none as these are not included in the accessibility tree >>

Actual: << Flags up the ID of an element that is not in the accessibility tree >>

Motivation: << We believe this is a false positive >>

<pre><code>
axe-core version: 4.3.5

Chrome 97.0.4692.99

</code></pre>

Found on https://www.argos.co.uk/search/red/?clickOrigin=searchbar:home:term:red 
<img width="965" alt="2022-01-27_11-52-08" src="https://user-images.githubusercontent.com/60389544/151353799-eccfe492-8b2d-40cb-b9ba-699f48fe09de.png">

**Comment by straker** [comment]
Thanks for the issue. So duplicate ids can be a problem even if the element associated with the id is hidden. This is because ARIA attributes can reference those elements even if they aren't in the accessibility tree.

For example, using `aria-labelledby` to point to an element that is hidden will still result in an accessible name:

```html
<button aria-labelledby="label-text"></button>
<div id="label-text" style="display: none">My button</div>
```

If there were multiple elements with the same if of `label-text`, then that would cause an accessibility problem (even if both were hidden in the accessibility tree).

**Comment by Brynman** [comment]
Hi @straker and thanks for your reply.

Using your example, if both the button element and the `<div>` which labels it are hidden, is this still an issue?

```
<button aria-labelledby="label-text" style="display: none"></button>
<div id="label-text" style="display: none">My button</div>
```

**Comment by WilcoFiers** [comment]
@Brynman Actually, you may be onto something here!

Are you saying that you have only one button with that `aria-labelledby` attribute, and that button is hidden? I was sort of assuming when I saw this case there would be multiple buttons, one visible; one hidden.

I suppose if there is only one button, while it's hidden that probably should be flagged as a `duplicate-id` error, rather than a `duplicate-id-active` error. That's still an error, but not as impactful. Can you explain in what situation you would have such a button? It seems to me like that's definitely going to create problems, just maybe not exactly in the state you've tested.

**Comment by Brynman** [comment]
Hi @WilcoFiers @straker...
`<button>` A controls `<input>` A.
`<button>` B controls `<input>` B.
`<button>` A and `<input>` B have the same ID, however
`<button>` A and `<input>` A are hidden when `<button>` B and `<input>` B are exposed, and vice versa.

Does this change anything?

**Comment by Brynman** [comment]
Cheers @WilcoFiers , will get someone to take a look. Thanks for taking the time to look into this. Much appreciated as always.

**Comment by WilcoFiers** [comment]
@Brynman The idea behind this rule is that there is no standard that says what browsers / assistive tech should do with duplicate IDs related to accessible names. You may be right that all assistive technologies behave the same, but that doesn't mean this can't change 6 months from now. The goal of this is to ensure things don't just work today, but that they continue to work.

Things like that do change. It's up to you if you want to risk that. It's probably not a big risk to take. Then again, resolving duplicate IDs isn't too much work either.

**Comment by karanshah10** [comment]
@Brynman I am experiencing the same issue. Can you teach me the solution that you have implemented. Thanks!

**Comment by Brynman** [comment]
> @Brynman I am experiencing the same issue. Can you teach me the solution that you have implemented. Thanks!

@karanshah10 I didn't fix this myself but basically we prefixed the second cloned wishlist ID with a random string to make it unique.

**Comment by WilcoFiers** [comment]
I don't think there's anything more to be done on this. Closing the issue.
